File: 外卖前端/src/api/shop.ts
import request from '@/utils/request'

export function getShopStatus() {
  return request.get('/admin/shop/status')
}

export function setShopStatus(status: number) {
  return request.put(`/admin/shop/status`, null, { params: { status } })
}

export function registerShop(data: {
  name: string
  ownerName: string
  phone: string
  address: string
  description?: string
}) {
  return request.post('/admin/shop/register', data)
}

export function getShopPage(params: { page: number; pageSize: number; name?: string; status?: number }) {
  return request.get('/admin/shop/page', { params })
}

export function approveShop(shopId: string) {
  return request.put(`/admin/shop/approve/${shopId}`)
}

export function rejectShop(shopId: string) {
  return request.put(`/admin/shop/reject/${shopId}`)
}

export function getShopByShopId(shopId: string) {
  return request.get(`/admin/shop/info/${shopId}`)
}

export function updateShop(shopId: string, data: {
  name?: string
  image?: string
  address?: string
  description?: string
  phone?: string
  ownerName?: string
}) {
  return request.put(`/admin/shop/info/${shopId}`, data)
}

export function setBusinessStatus(shopId: string, status: number) {
  return request.put(`/admin/shop/businessStatus`, null, { params: { shopId, status } })
}

File: java后端/sky-pojo/src/main/java/com/sky/entity/Shop.java
package com.sky.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 店铺
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Shop implements Serializable {

    private static final long serialVersionUID = 1L;

    private Long id;

    // 店铺编号（随机生成）
    private String shopId;

    // 店铺名称
    private String name;

    // 店主姓名
    private String ownerName;

    // 联系电话
    private String phone;

    // 店铺地址
    private String address;

    // 店铺图片
    private String image;

    // 店铺简介
    private String description;

    // 审核状态 0待审核 1已通过 2已拒绝 3已禁用
    private Integer status;

    // 营业状态 0打烊 1营业中
    private Integer businessStatus;

    private LocalDateTime createTime;

    private LocalDateTime updateTime;
}
